model/project_ref.go

```diff
@@ -1205,16 +1216,29 @@ func FindDownstreamProjects(project string) ([]ProjectRef, error) {
 
 // FindOneProjectRefByRepoAndBranchWithPRTesting finds a single ProjectRef with matching
 // repo/branch that is enabled and setup for PR testing.
-func FindOneProjectRefByRepoAndBranchWithPRTesting(owner, repo, branch string) (*ProjectRef, error) {
+func FindOneProjectRefByRepoAndBranchWithPRTesting(owner, repo, branch, calledBy string) (*ProjectRef, error) {
 	projectRefs, err := FindMergedEnabledProjectRefsByRepoAndBranch(owner, repo, branch)
 	if err != nil {
 		return nil, errors.Wrapf(err, "Could not fetch project ref for repo '%s/%s' with branch '%s'",
 			owner, repo, branch)
 	}
 	for _, p := range projectRefs {
-		if p.IsPRTestingEnabled() {
-			p.checkDefaultLogger()
-			return &p, nil
+		switch calledBy {
+		case patch.CalledManually:
+			if p.IsManualPRTestingEnabled() {
+				p.checkDefaultLogger()
+				return &p, nil
+			}
+		case patch.CalledAutomatically:
+			if p.IsAutoPRTestingEnabled() {
+				p.checkDefaultLogger()
+				return &p, nil
+			}
+		default:
+			if p.IsPRTestingEnabled() {
```

Comment:
Need to also consider this at 1260. A way to move this to a helper would be IsPRTestingEnabledByCaller(caller string) so you don't need to case each time you need to check

Reply:
What I mean is, rather than doing this switch statement here to determine which function to return, you can have IsPRTestingEnabledByCaller do:
```
case patch.ManualCaller:
	return p.IsManualPRTestingEnabled()
case patch.AutomatedCaller:
	return  p.IsAutoPRTestingEnabled() 
default:
	return p.IsPRTestingEnabled()
```
and then this function call just do
```
if  p.IsPRTestingEnabledByCaller(calledBy) { 
           p.checkDefaultLogger()
	   return &p, nil
```